From 4788988827bf4c0d996d3e1e3b6ac52a23b97941 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Tue, 9 Jan 2024 21:15:47 +0100
Subject: [PATCH] configure.in: use pkg-config to find libnet

libnet-config is deprecated:
https://github.com/libnet/libnet/commit/1ec4a6e7d1d23811f0a14ecb51e46489a0175727

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Upstream: unsent (upstream seems dead, no commit since 5 years
---
 configure.in | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/configure.in b/configure.in
index 7c3dd43..882ff86 100644
--- a/configure.in
+++ b/configure.in
@@ -176,12 +176,14 @@ AC_ARG_ENABLE(tcpkill,
 [ use_tcpkill="$enableval" ],
 [ use_tcpkill="no" ])
 
+PKG_PROG_PKG_CONFIG
 if test $use_tcpkill = yes; then
   AC_CHECK_LIB(net, libnet_init,,echo !!! error: tcpkill feature enabled but no libnet found; exit)
   USE_TCPKILL="1"
   EXTRA_OBJS="$EXTRA_OBJS tcpkill.o"
-  EXTRA_DEFINES="$EXTRA_DEFINES $(libnet-config --defines)"
-  EXTRA_LIBS="$EXTRA_LIBS $(libnet-config --libs)"
+  PKG_CHECK_MODULES(LIBNET, libnet)
+  EXTRA_DEFINES="$EXTRA_DEFINES $LIBNET_CFLAGS"
+  EXTRA_LIBS="$EXTRA_LIBS $LIBNET_LIBS"
 else
   USE_TCPKILL="0"
 fi
-- 
2.43.0

